src/main/java/org/folio/service/inventory/InventoryItemManager.java

```diff
@@ -93,69 +93,80 @@ public InventoryItemManager(RestClient restClient,
+      .map(this::extractEntities);
   }
 
   public Future<JsonObject> getItemRecordById(String itemId, boolean skipThrowNorFoundException, RequestContext requestContext) {
     RequestEntry requestEntry = new RequestEntry(INVENTORY_LOOKUP_ENDPOINTS.get(ITEM_BY_ID_ENDPOINT)).withId(itemId);
     return restClient.getAsJsonObject(requestEntry, skipThrowNorFoundException, requestContext);
   }
 
-  /**
-   * Returns list of requests for specified item.
-   *
-   * @param itemId id of Item
-   * @return future with list of requests
-   */
-  public Future<Integer> getNumberOfRequestsByItemId(String itemId, RequestContext requestContext) {
-    String query = String.format("(itemId==%s and status=\"*\")", itemId);
-    RequestEntry requestEntry = new RequestEntry(INVENTORY_LOOKUP_ENDPOINTS.get(REQUESTS))
-      .withQuery(query).withOffset(0).withLimit(0);
-    return restClient.getAsJsonObject(requestEntry, requestContext)
-      .map(json -> json.getInteger(TOTAL_RECORDS));
+  public Future<List<JsonObject>> getItemsByHoldingIdAndOrderLineId(String holdingId, String purchaseOrderLineId, RequestContext requestContext) {
+    String query = String.format("holdingsRecordId==%s and purchaseOrderLineIdentifier==%s", holdingId, purchaseOrderLineId);
+    return getItemRecordsByQuery(query, requestContext);
+  }
+
+  public Future<List<JsonObject>> getItemsByPoLineIdsAndStatus(List<String> poLineIds, String itemStatus, RequestContext requestContext) {
+    logger.debug("getItemsByStatus start");
+    List<Future<List<JsonObject>>> futures = StreamEx
+      .ofSubLists(poLineIds, MAX_IDS_FOR_GET_RQ_15)
+      .map(ids -> {
+        String query = String.format("status.name==%s and %s", itemStatus, HelperUtils.convertFieldListToCqlQuery(ids, InventoryItemManager.ITEM_PURCHASE_ORDER_LINE_IDENTIFIER, true));
+        return getItemRecordsByQuery(query, requestContext);
+      })
+      .toList();
+
+    return collectResultsOnSuccess(futures)
+      .map(lists -> StreamEx.of(lists).toFlatList(jsonObjects -> jsonObjects));
   }
 
-  /**
-   * Returns list of item records for specified query.
-   *
-   * @param query item records query
-   * @return future with list of item records
-   */
   private Future<List<JsonObject>> getItemRecordsByQuery(String query, RequestContext requestContext) {
     RequestEntry requestEntry = new RequestEntry(INVENTORY_LOOKUP_ENDPOINTS.get(ITEMS))
       .withQuery(query).withOffset(0).withLimit(Integer.MAX_VALUE);
     return restClient.getAsJsonObject(requestEntry, requestContext)
-      .map(response -> extractEntities(response, ITEMS));
+      .map(this::extractEntities);
+  }
+
+  public Future<Integer> getNumberOfRequestsByItemId(String itemId, RequestContext requestContext) {
+    String query = String.format("(itemId==%s and status=\"*\")", itemId);
+    RequestEntry requestEntry = new RequestEntry(INVENTORY_LOOKUP_ENDPOINTS.get(REQUESTS))
+      .withQuery(query).withOffset(0).withLimit(0);
```

Comment:
I think Limit is supposed to be Integer Max Value

Reply:
I asked this as well. And why current version is working (I didn't modified, just replaced)?

Reply:
@Saba-Zedginidze-EPAM please take a look is it expected that we use limit = 0 here as you cyrrently busy with requests integration

Reply:
As I know, this results in the record bodies being empty but the totalRecords shows how many entities were originally selected. I noticed this issue in my PR aswell and am currently modifying the code